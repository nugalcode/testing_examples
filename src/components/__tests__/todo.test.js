import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from '../todo';

// jest runs this after every test in this file completes
// if function returns a promise, Jest waits for that promise to resolve
// before continuing
// useful for cleaning up temporary state created by each test
afterEach(() => {
    cleanup();
})


test('should render non-completed todo', () => {
    const todo = {
        id: 1,
        title: 'wash dishes',
        completed: false,
    };

    render(<Todo todo={todo}/>)
    const todoElement = screen.getByTestId('todo-1');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('wash dishes');
    // check if element does NOT exist
    expect(todoElement.querySelector('s')).toBeNull();
})

test('should render completed todo', () => {
    const todo = {
        id: 2,
        title: 'wash car',
        completed: true,
    };

    render(<Todo todo={todo} />)
    const todoElement = screen.getByTestId('todo-2');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('wash car');
    // tests if todoElement has a child with a 's' tag
    expect(todoElement.querySelector('s')).toBeInTheDocument();
})

test('matches snapshot', () => {
    const todo = {
        id: 1,
        title: 'wash dishes',
        completed: false,
    };
    const tree = renderer.create(<Todo todo={todo} />).toJSON();
    expect(tree).toMatchSnapshot();

})