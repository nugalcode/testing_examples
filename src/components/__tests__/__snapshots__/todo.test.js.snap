// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="hi"
  data-testid="todo-1"
>
  <h1>
     
    wash dishes
     
  </h1>
</div>
`;
